src/pydot/core.py

```diff
@@ -1434,9 +1434,9 @@ def to_string(
             edge_src_set, edge_dst_set = list(
                 zip(*[obj["points"] for obj in edge_obj_dicts])
             )
-            edge_src_set, edge_dst_set = set(edge_src_set), set(edge_dst_set)  # type: ignore
+            edge_src_set, edge_dst_set = set(edge_src_set), set(edge_dst_set)
         else:
-            edge_src_set, edge_dst_set = set(), set()  # type: ignore
+            edge_src_set, edge_dst_set = set(), set()
```

Comment:
These, OTOH, make total sense.

Reply:
Oh, no they don't, because the variables are initially created from a `zip()` so they're tuples.

Best solution for this is probably...

```python
if edge_obj_dicts:
    edge_srcs, edge_dsts = list(
        zip(*[obj["points"] for obj in edge_obj_dicts])
    )
    edge_src_set, edge_dst_set = set(edge_srcs), set(edge_dsts)
else:
    edge_src_set, edge_dst_set = set(), set()
```

Reply:
Actually, wait, never mind. I always meant to fix this.

There's _no reason_ to keep separate lists of the two edge-endpoint sides.

These sets are only used for `suppress_disconnected` mode, which only cares if a node has no connections. It doesn't MATTER which endpoint of the edge it's on, so the endpoints can all be kept in one big list instead of dealing with two separate.

I'll put together a quick PR to fix that the right way.